lib/exchange.ts

```diff
@@ -0,0 +1,38 @@
+import { Contract, ContractInterface, Signer } from "ethers"
+import { ILogObj, Logger } from "tslog"
+
+const log: Logger<ILogObj> = new Logger({ hideLogPositionForProduction: true })
+export const approveClaimTicket = async (
+  signer: Signer,
+  abi: ContractInterface,
+  claimTicketId: number | string,
+) => {
+  const contract = new Contract(process.env.NEXT_PUBLIC_CRE8ORS_CLAIM_TICKET_ADDRESS, abi, signer)
+  try {
+    const tx = await contract.approve(
+      process.env.NEXT_PUBLIC_CRE8ORS_EXCHANGE_ADDRESS,
+      claimTicketId,
+    )
+    const res = await tx.wait()
+    return res
+  } catch (e) {
+    log.error(e)
+    return e
+  }
```

Comment:
bug: you are `catching` signature rejections here, so, in UX, user still progresses to `/success` screen.

suggestion: either
1. do not `try / catch` here, handle errors in caller
2. return an `isSuccess` variable

Reply:
done! removed the try catch here